unreleased_history/new_features/timeout_for_wait_for_compact_api.md

```diff
@@ -0,0 +1 @@
+Add timeout in microsecond option to `WaitForCompactOptions` to prevent `WaitForCompact()` from waiting indefinitely
```

Comment:
I think we should avoid implying that indefinite wait is a normal possibility. (Indefinite waiting should only happen in recurring recoverable errors, such as out-of-space.)

Reply:
I agree with you on this sentiment, but we indicated that enough stream of writes keeps flush and compactions going can make `WaitForCompact()` wait for very long time till the writes settle. This timeout could also help user stop waiting in this scenario and I wanted to include that.

How about

"Add timeout in microsecond option to `WaitForCompactOptions` to allow timely termination of prolonged waiting in scenarios like recurring recoverable errors, such as out-of-space situations or continuous write streams that sustain ongoing flush and compactions"?